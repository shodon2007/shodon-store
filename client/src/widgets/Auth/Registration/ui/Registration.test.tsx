import { screen } from '@testing-library/react';
import renderTestApp from 'src/app/tests/helpers/renderTestApp';
import '@testing-library/jest-dom/jest-globals';

describe('jest testing registration.test.tsx', () => {
	test('testing elements in document', () => {
		renderTestApp('/auth?tab=registration');

		const registrationPage = screen.getByTestId('registration-page');
		const registrationButton = screen.getByTestId('registration-button');
		const registrationInputs = screen.getAllByTestId('registration-input');

		expect(registrationPage).toBeInTheDocument();
		expect(registrationButton).toBeInTheDocument();
		expect(registrationInputs.length).toBe(2);
	});

	test('testing snapshopts', () => {
		renderTestApp('/auth?tab=registration');

		const registrationPage = screen.getByTestId('registration-page');

		expect(registrationPage).toMatchSnapshot();
	});

	test('testing headers', () => {
		const page1 = renderTestApp('/auth?tab=dfjksdjflsdfk');

		expect(page1.queryByTestId('registration-page')).not.toBeInTheDocument();
	});
});
