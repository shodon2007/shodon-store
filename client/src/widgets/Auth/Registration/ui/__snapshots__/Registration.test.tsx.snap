// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`jest testing registration.test.tsx testing snapshopts 1`] = `
<div
  class="inputs"
  data-testid="registration-page"
>
  <input
    class="Input "
    data-testid="registration-input"
    placeholder="login"
    type="text"
    value=""
  />
  <input
    class="Input "
    data-testid="registration-input"
    minlength="8"
    placeholder="password"
    required=""
    type="password"
    value=""
  />
</div>
`;
